pkg/services/user/userimpl/store.go

```diff
@@ -306,13 +305,11 @@ func (ss *sqlStore) loginConflict(ctx context.Context, sess *db.Session, login,
 	if exists {
 		return user.ErrUserAlreadyExists
 	}
-	if err := sess.Where("LOWER(email)=LOWER(?) OR LOWER(login)=LOWER(?)",
-		email, login).Find(&users); err != nil {
-		return err
-	}
-
-	if len(users) > 1 {
-		return &user.ErrCaseInsensitiveLoginConflict{Users: users}
+	if caseInsensitive {
```

Comment:
**main difference**

We are checking the caseInsensitive option before checking for the lowercasings